tests/components/case.list.spec.js

```diff
@@ -52,19 +55,43 @@ describe('CaseList', ()=> {
             expect(document.find('#editFormModal').prop('style').display).toEqual('block');
 
             expect(document.find('#country').at(0).prop('value')).toEqual('Canada');
-            expect(document.find('#city').at(0).prop('value')).toEqual('old-value');
+            expect(document.find('#city').at(0).prop('value')).toEqual('old-city');
 
             document.find('#editFormModal button#back').prop('onClick')();
             document.update();
             expect(document.find('#editFormModal').prop('style').display).toEqual('none');
         });
-        test('can be changed', ()=>{
+        test('city can be changed', ()=>{
             document.find('#edit-15').prop('onClick')();
             document.update();
-            let field = document.find('#city').at(0);
-            field.simulate('change', { target: { name:'respondent.city', value:'new-value' } });
+            let field = document.find('input#city').at(0);
+            field.simulate('change', { target: { name:'respondent.city', value:'new-city' } });
 
-            expect(cases[1].data.respondents[0].address.city).toEqual('new-value');
+            expect(cases[1].data.respondents[0].address.city).toEqual('new-city');
+        });
+        test('addressLine1 can be changed', ()=>{
+            document.find('#edit-15').prop('onClick')();
+            document.update();
+            let field = document.find('input#addressLine1').at(0);
+            field.simulate('change', { target: { name:'respondent.addressLine1', value:'new-addressLine1' } });
+
+            expect(cases[1].data.respondents[0].address.addressLine1).toEqual('new-addressLine1');
+        });
+        test('addressLine2 can be changed', ()=>{
+            document.find('#edit-15').prop('onClick')();
+            document.update();
+            let field = document.find('input#addressLine2').at(0);
+            field.simulate('change', { target: { name:'respondent.addressLine2', value:'new-addressLine2' } });
+
+            expect(cases[1].data.respondents[0].address.addressLine2).toEqual('new-addressLine2');
+        });
+        test('postalCode can be changed', ()=>{
+            document.find('#edit-15').prop('onClick')();
+            document.update();
+            let field = document.find('input#postalCode').at(0);
+            field.simulate('change', { target: { name:'respondent.postalCode', value:'new-postalCode' } });
```

Comment:
hmmm... technically, this value should be rejected, since postal code only allows A1A 1A1 format.  I wonder if this is a valid test now.

Reply:
The validation that you are pointing out is under onBlur event.
This test is validating onChange event which implements a different validation